charts/extended-ceph-exporter/templates/post-install-job.yaml

```diff
@@ -0,0 +1,131 @@
+              # can be run directly, instead of requiring the rook toolbox
+              CEPH_CONFIG="/etc/ceph/ceph.conf"
+              MON_CONFIG="/etc/rook/mon-endpoints"
+              KEYRING_FILE="/etc/ceph/keyring"
+
+              # create a ceph config file in its default location so ceph/rados tools can be used
+              # without specifying any arguments
+              write_endpoints() {
+                endpoints=$(cat ${MON_CONFIG})
+
+                # filter out the mon names
+                # external cluster can have numbers or hyphens in mon names, handling them in regex
+                # shellcheck disable=SC2001
+                mon_endpoints=$(echo "${endpoints}"| sed 's/[a-z0-9_-]\+=//g')
+
+                DATE=$(date)
+                echo "$DATE writing mon endpoints to ${CEPH_CONFIG}: ${endpoints}"
+                  cat <<EOF > ${CEPH_CONFIG}
+              [global]
+              mon_host = ${mon_endpoints}
+
+              [client.admin]
+              keyring = ${KEYRING_FILE}
+              EOF
+              }
+
+              # create the keyring file
+              cat <<EOF > ${KEYRING_FILE}
+              [${ROOK_CEPH_USERNAME}]
+              key = ${ROOK_CEPH_SECRET}
+              EOF
+
+              # write the initial config file
+              write_endpoints
+          env:
+            - name: ROOK_CEPH_USERNAME
+              valueFrom:
+                secretKeyRef:
+                  name: rook-ceph-mon
+                  key: ceph-username
+            - name: ROOK_CEPH_SECRET
+              valueFrom:
+                secretKeyRef:
+                  name: rook-ceph-mon
+                  key: ceph-secret
+            - name: CEPH_METRICS_RGW_HOST
+              valueFrom:
+                secretKeyRef:
+                 name: {{ include "extended-ceph-exporter.fullname" . }}
+                 key: CEPH_METRICS_RGW_HOST
+            - name: CEPH_METRICS_RGW_ACCESS_KEY
+              valueFrom:
+                secretKeyRef:
+                 name: {{ include "extended-ceph-exporter.fullname" . }}
+                 key: CEPH_METRICS_RGW_ACCESS_KEY
+            - name: CEPH_METRICS_RGW_SECRET_KEY
+              valueFrom:
+                secretKeyRef:
+                 name: {{ include "extended-ceph-exporter.fullname" . }}
+                 key: CEPH_METRICS_RGW_SECRET_KEY
+          volumeMounts:
+            - mountPath: /etc/ceph
+              name: ceph-config
+            - name: mon-endpoint-volume
+              mountPath: /etc/rook
+      containers:
+        - name: post-install-job-fetch-name
+          image: "{{ .Values.postInstallJob.image.repository }}:{{ .Values.postInstallJob.image.tag }}"
+          imagePullPolicy: {{ .Values.postInstallJob.image.pullPolicy }}
+          command:
+          - /bin/bash
+          - -c
+          - |
+            set -x
+            user=$(radosgw-admin user list | jq -r '.[0]')
```

Comment:
This needs to be checking the whole user list and not just the first user, using grep to check for `"extended-ceph-exporter"` should be good enough here.